### PR TITLE
Add support for adapt_framework -> grunt server-build:dev

### PR DESCRIPTION
**Pull request by taylortom**
Partial fix for #811

### CODE REVIEW

**Comment by taylortom** [comment]
Also see https://github.com/adaptlearning/adapt_framework/pull/734/

**Comment by brian-learningpool** [comment]
@taylortom, I really like this.  Source maps would be of major help to developers trying to debug client-side code.  The only thing I have issue with is using the `isProduction` config to toggle this.  This config is already used in determine whether the web application should run in production mode.

Perhaps a new course setting would be more applicable?

**Comment by taylortom** [comment]
Fine by me - are you thinking another config.json option? (e.g. `generateOutputSourcemap`)

**Comment by brian-learningpool** [comment]
Yes, either in Configuration Settings or Project Settings.

**Comment by taylortom** [comment]
@brian-learningpool this work should be done now

**Comment by taylortom** [comment]
TODO: test this in older versions of the framework to make sure the params don't cause any unexpected problems if not used.

**Comment by brian-learningpool** [comment]
@taylortom, this is really useful.  I've just been testing this locally and I think there's one small thing it's missing though.  When the `_generateSourcemap` value changes you you should trigger a flag to indicate that a full rebuild is required.  Have a look at the `rebuildCourse` event:

```
 app.emit('rebuildCourse', tenantId, courseId);
```

File: plugins/output/adapt/index.js [hunk 1]
```diff
@@ -215,14 +218,14 @@ AdaptOutput.prototype.publish = function (courseId, isPreview, request, response
...
+            { expand: true, cwd: path.join(FRAMEWORK_ROOT_FOLDER, Constants.Folders.AllCourses, tenantId, courseId, Constants.Folders.Build), src: ['**/*'] },
+          ]).finalize();
 
         } else {
           // No download required -- skip this step
           callback();
-        }        
-      }   
+        }
+      },
```
Comment (dennis-learningpool) [inline plugins/output/adapt/index.js]: usually wary of trailing commas in a JS array like this - is it ok?

Comment (taylortom) [inline plugins/output/adapt/index.js]: Think I ran the jscs grunt task at some point which may have added this.

**Comment by taylortom** [comment]
+1 for @brian-learningpool's changes

**Comment by brian-learningpool** [comment]
+1 for yours too! :+1:

**Comment by dennis-learningpool** [comment]
+1 for everyone!

**Pull request merged**